llvm/lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -8052,18 +8065,30 @@ Error ModuleSummaryIndexBitcodeReader::parseEntireSummary(unsigned ID) {
         }
         MIBs.push_back(MIBInfo(AllocType, std::move(StackIdList)));
       }
-      std::vector<uint64_t> TotalSizes;
-      // We either have no sizes or NumMIBs of them.
-      assert(I == Record.size() || Record.size() - I == NumMIBs);
+      // We either have nothing left or at least NumMIBs context size info
+      // indices left.
+      assert(I == Record.size() || Record.size() - I >= NumMIBs);
+      std::vector<std::vector<unsigned>> AllContextSizeIndices;
       if (I < Record.size()) {
         MIBsRead = 0;
-        while (MIBsRead++ < NumMIBs)
-          TotalSizes.push_back(Record[I++]);
+        while (MIBsRead++ < NumMIBs) {
+          unsigned NumContextSizeInfoEntries = Record[I++];
+          assert(Record.size() - I >= NumContextSizeInfoEntries);
+          std::vector<unsigned> ContextSizeIndices;
```

Comment:
`ContextSizeIndices.reserve( NumContextSizeInfoEntries)` since we know how many we will insert?